### PR TITLE
initial testing of gh-action trigger on PRs with automatic token

### LINKED ISSUE #9: Get running on github actions
Do this first for minimal case, then add features.

### PR DESCRIPTION
**Pull request by yidilozdemir**
Draft PR so I could test the building and deployment in action. This PR only contains a trigger to regenerate the pages. After confirming the backbone works, will add the building components

### CODE REVIEW

**Comment by yidilozdemir** [comment]
Confirm it works! Deployed from `gh-pages`. Discovered a massive potential problem when I initially set the base and target directory of deployment to `master`: this should never be done! It overwrote the directory to only contain a config file that contains the hash for deployment.

**Comment by yidilozdemir** [comment]
Confirming it works, ended up using a Github action that is more straightforward 
`uses: JamesIves/github-pages-deploy-action@v4.4.1`

**Review by thompson318 — APPROVED** [review]

**Review by mxochicale — APPROVED** [review]
Hi Idil, 
All looks good to me. Perhaps sometime to think for coming task. 
Do we want to improve python code quality with `pylint`? Feel free to open a new issue. 

Thanks, Miguel

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,4 @@
+printf '%s\n' "$github_token" >github_token
 #step 1 search for relevant packages on pypi and githib
```
[the remark this answers is not in the document]
Comment (yidilozdemir) [reply Makefile:2]: I also just realised, i'll remove that one liner. Thank you for the suggestion

**Pull request merged**